File: leetcode/88.py
"""
    File: 88.py
    Title: Merge Sorted Array
    Difficulty: Easy
"""

from typing import List


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p1, p2 = m - 1, n - 1
        for p in range(m + n - 1, -1, -1):
            if p1 < 0:
                nums1[p] = nums2[p2]
                p2 -= 1
            elif p2 < 0:
                nums1[p] = nums1[p1]
                p1 -= 1
            elif nums1[p1] > nums2[p2]:
                nums1[p] = nums1[p1]
                p1 -= 1
            else:
                nums1[p] = nums2[p2]
                p2 -= 1

File: leetcode/88.go
// File: 88.go
// Title: Merge Sorted Array
// Difficulty: Easy

package leetcode

func merge(nums1 []int, m int, nums2 []int, n int) {
	i := m - 1
	j := n - 1
	k := m + n - 1
	for {
		if i < 0 && j < 0 {
			break
		} else if i < 0 {
			nums1[k] = nums2[j]
			j -= 1
		} else if j < 0 {
			nums1[k] = nums1[i]
			i -= 1
		} else if nums1[i] > nums2[j] {
			nums1[k] = nums1[i]
			i -= 1
		} else {
			nums1[k] = nums2[j]
			j -= 1
		}
		k -= 1
	}
}
